src/pykeen/models/unimodal/trans_h.py

```diff
@@ -90,47 +90,39 @@ def __init__(
         self.scoring_fct_norm = scoring_fct_norm
 
         # embeddings
-        self.normal_vector_embeddings = get_embedding(
+        self.normal_vector_embeddings = Embedding.init_with_device(
             num_embeddings=triples_factory.num_relations,
             embedding_dim=embedding_dim,
             device=self.device,
+            # Normalise the normal vectors by their l2 norms
+            constrainer=functional.normalize,
         )
 
-    def _reset_parameters_(self):  # noqa: D102
-        for emb in [
-            self.entity_embeddings,
-            self.relation_embeddings,
-            self.normal_vector_embeddings,
-        ]:
-            emb.reset_parameters()
-        # TODO: Add initialization
-
     def post_parameter_update(self) -> None:  # noqa: D102
-        # Make sure to call super first
         super().post_parameter_update()
+        self.normal_vector_embeddings.post_parameter_update()
 
-        # Normalise the normal vectors by their l2 norms
-        functional.normalize(
-            self.normal_vector_embeddings.weight.data,
-            out=self.normal_vector_embeddings.weight.data,
-        )
+    def _reset_parameters_(self):  # noqa: D102
+        super()._reset_parameters_()
+        self.normal_vector_embeddings.reset_parameters()
+        # TODO: Add initialization
 
     def regularize_if_necessary(self) -> None:
         """Update the regularizer's term given some tensors, if regularization is requested."""
         # As described in [wang2014], all entities and relations are used to compute the regularization term
         # which enforces the defined soft constraints.
         super().regularize_if_necessary(
-            self.entity_embeddings.weight,
-            self.normal_vector_embeddings.weight,
-            self.relation_embeddings.weight,
+            self.entity_embeddings(indices=None),
```

Comment:
On first look, this seems a bit misleading but then i re-read the signature of `regularize_if_necessary()` and it's only reading some tensors, and not writing to them. What's the FIXME though?

Reply:
I am not sure why there is a FIXME 😅